### ISSUE TITLE
[Discussion] Gab and Copyright

### ISSUE THREAD

**Issue by koyuawsmbrtn** [tootsuite/mastodon#11275]
AGPL has no protection against DCMA so what about filing mass DMCA requests to stop them to use our code? Would be a good thing.

**Comment by noirscape** [comment]
That's not how the AGPL works.

The AGPL demands that Gab:
- states the changes they made to to Mastodons source code
- releases the full source code under the AGPL (or a license that provides similar protections) so that anyone can deploy or run their own instance.
- and this applies to server and client parts (why it's the AGPL)

Gab iirc has the source code public, so they're in full compliance with the AGPL, so there is no copyright infringement whatsoever.

**Comment by koyuawsmbrtn** [comment]
> That's not how the AGPL works.
> 
> The AGPL demands that Gab:
> 
>     * states the changes they made to to Mastodons source code
> 
>     * releases the full source code under the AGPL (or a license that provides similar protections) so that anyone can deploy or run their own instance.
> 
>     * and this applies to server and client parts (why it's the AGPL)
> 
> 
> Gab iirc has the source code public, so they're in full compliance with the AGPL, so there is no copyright infringement whatsoever.

I was talking about the right to pull off my own code and code from others if they choose to. Dropping a DMCA request is a more radical way to force them to pull it off.

**Comment by noirscape** [comment]
You can't do that. It's called estoppel.

At it's simplest, that means you can't go back on your word. Once your code is AGPL, it will always be AGPL and anyone can use it.

**Comment by koyuawsmbrtn** [comment]
> You can't do that. It's called estoppel.
> 
> At it's simplest, that means you can't go back on your word. Once your code is AGPL, it will always be AGPL and anyone can use it.

But Stallman said so

**Comment by noirscape** [comment]
I assume you are reliant on the lulz dot com article.

To paraphrase the article, the problem the article describes only applies to the GPLv2. The GPLv3 fixes the particular loophole that could be used for this situation.

Mastodon is under the AGPLv3, not the AGPLv2.

**Comment by koyuawsmbrtn** [comment]
Ok, let me rephrase it: If I go and slap you it was me and not someone else, the same applies to my own work. If I made some work it's mine and not someone else. Done.

**Comment by noirscape** [comment]
Of course. Gab is required to credit you and follow the terms you set for your code (in this case, the AGPLv3). This is something that is already done, either through git commit logs or through the AUTHORS.md file.

You can however not demand they stop using your code since you already gave them permission to do so when you made the code _available_ under the AGPLv3. That is covered/protected by estoppel.

Don't get me wrong, I dislike gab a lot and wish they wouldn't be using Mastodon code either. But there is no legal grounds to DMCA them as long as they follow the AGPL.

**Comment by koyuawsmbrtn** [comment]
> Of course. Gab is required to credit you and follow the terms you set for your code (in this case, the AGPLv3). This is something that is already done, either through git commit logs or through the AUTHORS.md file.
> 
> You can however not demand they stop using your code since you already gave them permission to do so when you made the code _available_ under the AGPLv3. That is covered/protected by estoppel.
> 
> Don't get me wrong, I dislike gab a lot and wish they wouldn't be using Mastodon code either. But there is no legal grounds to DMCA them as long as they follow the AGPL.

Ok, so now we're talking. They removed everyone from the AUTHORS.md file and the whole commit log is not in there.

**Comment by Gargron** [comment]
> Of course. Gab is required to credit you and follow the terms you set for your code (in this case, the AGPLv3). This is something that is already done, either through git commit logs or through the AUTHORS.md file.

Interestingly, Gab has erased all old commit messages in their fork, and replaced AUTHORS.md with their own text.

**Comment by noirscape** [comment]
Then it depends on if Mastodon has clause 7b and 7c active.

7b forbids removing credit from contributors.

7c forbids misrepresentation of the original source code.

If these are active, Gab is not legally allowed to remove AUTHORS.md or the commit logs.

**Comment by koyuawsmbrtn** [comment]
> Then it depends on if Mastodon has clause 7b and 7c active.
> 
> 7b forbids removing credit from contributors.
> 
> 7c forbids misrepresentation of the original source code.
> 
> If these are active, Gab is not legally allowed to remove AUTHORS.md or the commit logs.

![grafik](https://user-images.githubusercontent.com/40538332/60928340-4f00b880-a2ad-11e9-9f0a-40ef7b3627a2.png)

**Comment by noirscape** [comment]
The text above states these are optional clauses which means you have to explicitly stated you're using them.

**Comment by koyuawsmbrtn** [comment]
> The text above states these are optional clauses which means you have to explicitly stated you're using them.

Then let's wait for @Gargron to confirm this

**Comment by Gargron** [comment]
What do you mean by I have to explicitly state we're "using them"? Using them how? Do I just say "yeah, that's cool" and that makes them active?

**Comment by noirscape** [comment]
I'm not a lawyer so I'm not entirely sure, but the way I always added them is by adding them to the short license header in files.

**Comment by nightpool** [comment]
We'd have to relicense Mastodon to add any additional terms, which would take a lot of work tracking down old contributors.

However, there are other requirements that apply to all modifications of AGPLv3 source code:

1. that the program displays an "appropriate copyright notice"—since mastodon project contributors own the vast majority of the copyright on the work, this would theoretically include mentioning them. This is part of the "Appropriate Legal Notices" that have to be included in user interface.
2. that the program carries "prominent notices stating that you modified it, and giving a relevant date."

Does Gab do either of these things?

**Comment by Gargron** [comment]
Frankly I see no benefits to discussing this in public, if legal action is possible.

**Comment by nightpool** [comment]
Fair.
